From 5061a5a47f460614fdfc52f451bb5d3c59710281 Mon Sep 17 00:00:00 2001
From: Tom Lendacky <thomas.lendacky@amd.com>
Date: Tue, 22 Jul 2025 16:59:44 -0500
Subject: [PATCH 2/6] MdePkg: Add the COHERENCY_SFW_NO CPUID bit field

RH-Author: Luigi Leonardi <None>
RH-MergeRequest: 120: OvmfPkg/MemEncryptSevLib: Evict cache lines during SNP memory validation
RH-Jira: RHEL-125104
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Commit: [2/4] a18029a9c5d77d95fc4321099d693ee6c48d20e2

Update the CPUID 0x8000001F EBX definition to add the COHERENCY_SFW_NO
bit field. The COHERENCY_SFW_NO bit is used to indicate that the SEV-SNP
cache coherency mitigation is not needed.

Signed-off-by: Tom Lendacky <thomas.lendacky@amd.com>
---
 MdePkg/Include/Register/Amd/Cpuid.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/MdePkg/Include/Register/Amd/Cpuid.h b/MdePkg/Include/Register/Amd/Cpuid.h
index fdcbc475d5..0313ee2789 100644
--- a/MdePkg/Include/Register/Amd/Cpuid.h
+++ b/MdePkg/Include/Register/Amd/Cpuid.h
@@ -701,9 +701,14 @@ typedef union {
     UINT32    ReducedPhysBits : 5;
 
     ///
-    /// [Bit 31:12] Reserved
+    /// [Bit 30:12] Reserved
     ///
-    UINT32    ReservedBits    : 21;
+    UINT32    ReservedBits    : 20;
+
+    ///
+    /// [Bit 31] SEV-SNP Cache Coherency Mitigation is not needed
+    ///
+    UINT32    CoherencySfwNo  : 1;
   } Bits;
   ///
   /// All bit fields as a 32-bit value
-- 
2.50.1

